### PR TITLE
 Add shapeBorder option on App Bar

### PR DESCRIPTION
**Pull request by lsaudon**
I added borderRadius for a project.

### CODE REVIEW

**Comment by Hixie** [comment]
@lsaudon Thanks for your contribution! This will need a test to make sure we don't accidentally regress your change later, though.

cc @HansMuller @willlarche for review

**Comment by willlarche** [comment]
Please add screenshots when you get a moment.

**Comment by lsaudon** [comment]
`BorderRadius.only(
      bottomRight: Radius.circular(15.0),
      bottomLeft: Radius.circular(15.0),
    )`

Simple Appbar
capture d ecran 2018-10-03 a 10 08 05

And with bottom
capture d ecran 2018-10-03 a 10 09 52

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// Defaults to 4, the appropriate elevation for app bars.
   final double elevation;
 
+  /// The material's shape as well its shadow.
+  ///
+  /// If shape is non null, the [borderRadius] is ignored and the material's
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart]: There's no borderRadius here

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
...
+  /// clip boundary and shadow are defined by the shape.
+  ///
+  /// A shadow is only displayed if the [elevation] is greater than
+  /// zero.
+  final ShapeBorder shapeBorder;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart]: To be consistent with the button classes, Material, etc, this property should be called "shape"

File: packages/flutter/lib/src/material/app_bar.dart [hunk 3]
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// Defaults to 4, the appropriate elevation for app bars.
   final double elevation;
 
+  /// The material's shape as well its shadow.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart]: Explain what the shape means in terms of the app bar

File: packages/flutter/lib/src/material/app_bar.dart (hunk 1 above)
Comment (lsaudon) [reply packages/flutter/lib/src/material/app_bar.dart]: For the commentary I was inspired by the comment for the shape property in material.dart.

File: packages/flutter/lib/src/material/app_bar.dart (hunk 2 above)
Comment (lsaudon) [reply packages/flutter/lib/src/material/app_bar.dart]: Yes,  this property should be called "shape".

File: packages/flutter/lib/src/material/app_bar.dart (hunk 3 above)
Comment (lsaudon) [reply packages/flutter/lib/src/material/app_bar.dart]: It's the shapeborder. 
I got the wrong name.

Comment (HansMuller) [reply packages/flutter/lib/src/material/app_bar.dart]: After you've fixed things up and added a test, use git push to update this pull-request.

**Comment by willlarche** [comment]
So, I can't see from the screenshots but I assume the top corners are rounded too?

**Comment by lsaudon** [comment]
It's a shapeborder. Yes, the top corners can be rounded too. https://docs.flutter.io/flutter/painting/ShapeBorder-class.html

**Comment by willlarche** [comment]
Great! I don't think I'd recommend shaping the lower corners but we shouldn't forbid it. And the top ones are shaped in lots of apps these days.

**Comment by willlarche** [comment]
So, no objections from Material product. Code up to @Hixie and @HansMuller to LGTM.

**Comment by lsaudon** [comment]
I'm sorry, I'm sorry.
I can't do the tests. 
I have a lot of package errors with flutter and art in the project.

**Comment by lsaudon** [comment]
Is everything okay?

File: packages/flutter/test/material/app_bar_test.dart [hunk 4]
```diff
@@ -1422,4 +1422,26 @@ void main() {
...
+      ),
+    );
+
+    final Finder appBarFinder = find.byType(AppBar);
+
+    AppBar getAppBarWidget() {
+      return tester.widget<AppBar>(appBarFinder);
+    }
+
+    expect(getAppBarWidget().shape, roundedRectangleBorder);
```
Comment (HansMuller) [inline packages/flutter/test/material/app_bar_test.dart]: Checking that the AppBar's shape parameter correctly initializes its shape property is useful, however you should also verify that the the AppBar Material widget's shape property is set.

You may find https://docs.flutter.io/flutter/flutter_test/CommonFinders/descendant.html to be useful for this.

**Review by HansMuller — CHANGES_REQUESTED** [review]
Just a reminder: waiting on a small test change.

**Review by HansMuller — APPROVED** [review]

**Comment by HansMuller** [comment]
I will land this when the tests have finished.

**Pull request merged**